tests/std/tests/P0896R4_common_iterator/env.lst

```diff
@@ -1,4 +1,4 @@
 # Copyright (c) Microsoft Corporation.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-RUNALL_INCLUDE ..\strict_concepts_matrix.lst
+RUNALL_INCLUDE ..\strict_concepts_20_matrix.lst

```

Comment:
`P0896R4_common_iterator/test.cpp` includes `<range_algorithm_support.hpp>` which includes `<ranges>`. I'm confused as to why this is using a C++20 matrix instead of a C++latest matrix. Shouldn't all of the P0896R4 tests be using C++latest? (Applies to all of those files, not repeating this comment.)

Reply:
In practice, we don't turn off all of Ranges in C++20 mode. Doing so would either cripple the STL by removing `iterator_traits`  and everything that needs it _or_ revert to the C++17 implementation which would likely cause regressions when we re-enable the C++20 implementation later. For example, `<span>` wouldn't work in C++20 mode, and delaying the deprecation of `move_iterator::operator->` would regress behavior when we later enable things in C++20 mode. Regressions in `/std:c++20` behavior are to be avoided at all costs.

What we actually disable are the `view` concept and the physical contents of `<ranges>`, which includes all of the range factories and range adaptors. These are the parts of Ranges affected by the three ABI-breaking WG21 proposals. This properly defends against breaking changes to C++20 but leaves the rest of the C++20 STL functioning despite any reliance on Concepts and or machinery from the Ranges implementation while minimizing the risk of regressions when these things light up in `/std:c++20` some day.

`<range_algorithm_support.hpp>` uses components that are specified to be defined in `<ranges>`, but in practice are defined in `<xutility>` and `<iterator>` so as to be available to `<algorithms>`. I'll add a direct `<iterator>` include here to avoid relying on transitive includes in C++20 mode when `<ranges>` is empty, but this header will be relying on unlawful carnal knowledge of our standard headers for the interim.